### PR TITLE
Guac 53 api link

### PR DESCRIPTION
**Pull request by Pot4ssium**
Added API link to sync inventory.

### CODE REVIEW

File: application/frontend/src/game/main.ts
Comment (mosguinz) [inline application/frontend/src/game/main.ts:1]: What in the mother of hardcoding is this shit?

**Review by mosguinz — APPROVED** [review]
Whatever, shit works for now. Approved.

File: application/frontend/src/game/main.ts [hunk 1]
```diff
@@ -26,7 +26,10 @@ const config: Phaser.Types.Core.GameConfig = {
 }
 
 const StartGame = (parent: string) => {
-  return new Game({ ...config, parent })
```
Comment (ArjunS132) [inline application/frontend/src/game/main.ts:29]: Make this 1 line

**Comment by Pot4ssium** [comment]
YOOOO WE'RE LOCKED IN

File: application/frontend/src/game/scenes/InventoryMenu.ts [hunk 2]
```diff
@@ -1,202 +1,318 @@
...
+        reinforced_iron_plate: "Reinforced Iron Plate",
+    }
+
+    private floatingTextOffsetY = 20 // Y offset for overlapping texts
+    private activeFloatingTexts: Phaser.GameObjects.Text[] = [] // Store active floating texts to manage overlaps
+
+    constructor() {
+        super({ key: "InventoryMenu" })
+        // TODO: Properly add syncing with the backend
+        const resourceMap = ["iron_ore",
```
Comment (ArjunS132) [inline application/frontend/src/game/scenes/InventoryMenu.ts:45]: Set up a better sync

File: application/frontend/src/game/scenes/InventoryMenu.ts [hunk 3]
```diff
@@ -1,202 +1,318 @@
...
+            "copper_ingot",
+            "concrete",
+            "iron_plate",
+            "copper_plate",
+            "iron_rod",
+            "screws",
+            "wire",
+            "cable",]
+
+        ResourcesService.readResources().then((response) => {
```
Comment (ArjunS132) [inline application/frontend/src/game/scenes/InventoryMenu.ts:146]: remove this? The sync should be happening off the constructor

File: application/frontend/src/game/scenes/InventoryMenu.ts [hunk 4]
```diff
@@ -1,202 +1,318 @@
...
+        } as ResourceBase;
+        let data = []
+        data.push(res)
+        let body = {
+            requestBody: data
+
+        } as TDataSetResources;
+
+        ResourcesService.setResources(body);
+        // const token =  localStorage.getItem("access_token");
```
Comment (ArjunS132) [inline application/frontend/src/game/scenes/InventoryMenu.ts:265]: Remove comments

File: application/frontend/src/game/scenes/InventoryMenu.ts [hunk 5]
```diff
@@ -1,202 +1,318 @@
...
 
-      return true
+    // Method to deduct resources
+    deductFromInventory(resource: string, amount: number): boolean {
+        if (this.inventory[resource] && this.inventory[resource].count >= amount) {
+            this.inventory[resource].count -= amount
+            this.inventory[resource].textObject!.setText(
+                `${this.inventory[resource].count}`,
+            )
+            const resourceMap = ["iron_ore",
```
Comment (ArjunS132) [inline application/frontend/src/game/scenes/InventoryMenu.ts:285]: set up this map as a const and bring in data from the backend directly

**Review by ArjunS132 — APPROVED** [review]
Approved for now, but needs heavy refactoring

**Pull request merged**